allensdk/brain_observatory/behavior/session_apis/data_transforms/behavior_data_xforms.py

```diff
@@ -279,8 +283,11 @@ def get_metadata(self) -> Dict[str, Any]:
             "LabTracks_ID": self.get_external_specimen_name(),
             "full_genotype": self.get_full_genotype(),
             "behavior_session_uuid": bs_uuid,
-            "foraging_id": self.foraging_id,
-            "behavior_session_id": self.behavior_session_id,
-            "behavior_training_id": self.behavior_training_id,
+            "foraging_id": self.get_foraging_id(),
+            "behavior_session_id": self.get_behavior_session_id()
         }
+        if hasattr(self, "ophys_experiment_ids"):
```

Comment:
This logic is difficult to reason about. When will it have this property and when will it not?

Note for future improvement: we are referencing child properties and methods in the base class. For example `get_full_genotype` is referenced here, but is defined in a subclass of a subclass (BehaviorOphysDataXforms -> BehaviorOphysJsonApi). If it is being referenced here, it should be defined in a parent class, not in a subclass.

Reply:
I definitely agree that further improvements can be made to the how these API classes are organized. My original philosophy when refactoring was to have the DataXform classes be basically like data processing Mixin classes while stuff like the LimsApi or the JsonApi classes would solely be concerned with collecting 'raw' (in the sense that they haven't be processed into final desired forms) data/metadata.